### PR TITLE
[2/N] Implements workflow repo

### PR DESCRIPTION
**Pull request by saurav-c**
## Describe your changes and why you are making these changes
This PR implements the workflow repo.

### Key Decisions
- Each table's table name and column names are placed in the relevant model file. Ex: `models/workflow.go`. This is how the implementation in `repos` will have access to the column and table names.
- A helper method is used internally for querying one or multiple models. Ex: `getWorkflows` and `getWorkflow`

## Related issue number (if any)
ENG 1901 and 1902

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/golang/lib/repos/sqlite/workflow.go [hunk 1]
```diff
@@ -0,0 +1,220 @@
+package sqlite
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aqueducthq/aqueduct/lib/collections/utils"
```
Comment (saurav-c) [inline src/golang/lib/repos/sqlite/workflow.go:8]: TODO: Anything in `collections/utils` should be moved to `repos/utils` https://linear.app/aqueducthq/issue/ENG-1934/move-utils-in-collections-to-repos-as-necessary

File: src/golang/lib/repos/sqlite/workflow.go [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+	// Get the last run DAG by getting the max created_at timestamp for all workflow DAGs associated with each
+	// workflow in the organization.
+
+	// We want to return 1 row for each workflow, so we use a LEFT JOIN between the workflow_dag
+	// and workflow_dag_result tables. A LEFT JOIN outputs all rows in the left table even if there
+	// is no match with a row in the right table. If there is no match, the columns of the right table
+	// are NULL.
+	// This means that `last_run_at` and `status` in the query output can be NULL.
+	query := `
+		WITH workflow_results AS
```
Comment (saurav-c) [inline src/golang/lib/repos/sqlite/workflow.go:70]: @eunice-chan For this query all I did was add indenting for each clause, ex `SELECT`, `FROM`, etc. I did not improve any of the table name aliasing, as that can be done as part of a later PR. Let me know if you have any other ideas for improving readability for SQL queries.

**Review by eunice-chan — COMMENTED** [review]
Does this PR combined with the eng-1900 PR do everything required to migrate the workflow repo (except for the TODO you wrote in the comment)? How do we test that it's working?

**Comment by saurav-c** [comment]
@eunice-chan I'm working on a PR rn to setup the integration tests. We can merge these PRs once the tests have been implemented and are passing. 

The final PR for each collection will be to replace the existing reader and writers used by each server Handler with these new repo objects.

**Pull request merged**